lib/devTools/CodePipeline.ts

```diff
@@ -0,0 +1,81 @@
+import { LinuxBuildImage, PipelineBuildAction, PipelineBuildActionProps, PipelineProject } from '@aws-cdk/aws-codebuild';
+import { GitHubSourceAction, Pipeline, PipelineProps } from '@aws-cdk/aws-codepipeline';
+import { Role, ServicePrincipal } from '@aws-cdk/aws-iam';
+import { Construct, SSMParameterProvider } from '@aws-cdk/cdk';
+
+export interface Props extends PipelineProps {
+  githubOwner: string
+  repoName: string
+  ssmGithubTokenName: string
+  codeBuildRolePolicy: string
+}
+
+export class GithubNodePipeline extends Construct {
+
+  private codeBuildRole: Role;
+  private pipeline: Pipeline;
+
```

Comment:
This is more a general suggestion than a comment, making it here for examples sake. 

You may want to expose some underlying references to things - eg, pipeline ARN. This can be useful if someone wants to pass the ARN to something, like an SNS and CloudWatchEvent trigger. Most of the higher level constructs in CDK expose a couple of things to make composing a little more convenient. 

The bucket construct is worth taking a lot at because it's reasonably simple.

https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/aws-s3/lib/bucket.ts